tests/functional/examples/tokens/test_erc1155.py

```diff
@@ -301,56 +295,66 @@ def test_mint_one_burn_one(erc1155, w3, assert_tx_failed):
     assert erc1155.balanceOf(owner, 25) == 1
 
     # try and burn an item we don't control
-    assert_tx_failed(lambda: erc1155.burn(25, 1, transact={"from": a3}))
+    with tx_failed():
+        erc1155.burn(25, 1, transact={"from": a3})
 
     # burn an item that contains something we don't own
-    assert_tx_failed(lambda: erc1155.burn(595, 1, transact={"from": a1}))
+    with tx_failed():
+        erc1155.burn(595, 1, transact={"from": a1})
 
     # burn ah item passing a higher amount than we own
-    assert_tx_failed(lambda: erc1155.burn(25, 500, transact={"from": a1}))
+    with tx_failed():
+        erc1155.burn(25, 500, transact={"from": a1})
 
     erc1155.burn(25, 1, transact={"from": owner})
 
     assert erc1155.balanceOf(owner, 25) == 0
 
 
-def test_mint_batch_burn_batch(erc1155, w3, assert_tx_failed):
+def test_mint_batch_burn_batch(erc1155, w3, tx_failed):
     owner, a1, a2, a3, a4, a5 = w3.eth.accounts[0:6]
     # mint NFTs 11-20
 
     # check the balance
     assert erc1155.balanceOfBatch([a3, a3, a3], [11, 12, 13]) == [1, 1, 1]
 
     # try and burn a batch we don't control
-    assert_tx_failed(lambda: erc1155.burnBatch([11, 12], [1, 1]))
+    with tx_failed():
+        erc1155.burnBatch([11, 12], [1, 1])
 
     # ids and amounts array length not matching
-    assert_tx_failed(lambda: erc1155.burnBatch([1, 2, 3], [1, 1], transact={"from": a1}))
+    with tx_failed():
+        erc1155.burnBatch([1, 2, 3], [1, 1], transact={"from": a1})
 
     # burn a batch that contains something we don't own
-    assert_tx_failed(lambda: erc1155.burnBatch([2, 3, 595], [1, 1, 1], transact={"from": a1}))
+    with tx_failed():
+        erc1155.burnBatch([2, 3, 595], [1, 1, 1], transact={"from": a1})
 
     # burn a batch passing a higher amount than we own
-    assert_tx_failed(lambda: erc1155.burnBatch([1, 2, 3], [1, 500, 1], transact={"from": a1}))
+    with tx_failed():
+        erc1155.burnBatch([1, 2, 3], [1, 500, 1], transact={"from": a1})
 
     # burn existing
     erc1155.burnBatch([11, 12], [1, 1], transact={"from": a3})
 
     assert erc1155.balanceOfBatch([a3, a3, a3], [11, 12, 13]) == [0, 0, 1]
 
     # burn again, should revert
-    assert_tx_failed(lambda: erc1155.burnBatch([11, 12], [1, 1], transact={"from": a3}))
+    with tx_failed():
+        erc1155.burnBatch([11, 12], [1, 1], transact={"from": a3})
 
-    assert lambda: erc1155.balanceOfBatch([a3, a3, a3], [1, 2, 3]) == [0, 0, 1]
+    assert erc1155.balanceOfBatch([a3, a3, a3], [1, 2, 3]) == [0, 0, 0]
```

Comment:
hmm, interesting catch

Reply:
yeah, good catch, will check the rest of the codebase that we don't have such uncalled lambdas elsewhere